resources/type-system/upper-lower-bounds.md

```diff
@@ -81,6 +85,34 @@ We define the upper bound of two types T1 and T2 to be **UP**(`T1`,`T2`) as foll
 - **UP**(`T1`, `T2`) = `T2` if **BOTTOM**(`T1`)
 - **UP**(`T1`, `T2`) = `T1` if **BOTTOM**(`T2`)
 
+- **UP**(`X1 extends B1`, `T2`) =
```

Comment:
Do we need to define what `X extends B` means, "a type variable `X` with bound `B`"?

It's not the syntax we use for type variables otherwise.

Reply:
We have this notation in several documents (at least the NNBD feature specification and this document). I haven't used it in the language specification (the update with null safety replaces it by an explicit introduction of the bound), so it will presumably be replaced like that as various feature specifications are integrated into the language specification. Hence, there's no need to change it here.

Reply:
True, it is inconsistent that we write `X extends B` to indicate that a certain type denoted by `X` is a type variable and its declaration specifies `B` as the bound, and then we use `X & B2` to denote a promoted type variable. 

However, I don't think we have to worry about this because it will be reworded to introduce the bound explicitly in natural language ('Assume that `X` is a type variable with bound `B`') and then we'll just use `X` in the rules.